focus/internals/src/lib/project_cache/mod.rs

```diff
@@ -333,58 +403,87 @@ impl<'cache> ProjectCache<'cache> {
     /// Generate and push sharded project cache data.
     pub fn generate_and_push(
         &self,
-        keys: &Vec<NamespacedKey>,
+        generation_result: &GenerationResult,
         build_graph_hash: &Vec<u8>,
         shard_index: usize,
         shard_count: usize,
     ) -> anyhow::Result<()> {
-        let mut items = Vec::<(NamespacedKey, Value)>::new();
-        for key in keys {
+        let mandatory_items = {
+            let mut mandatory_items = BTreeMap::<String, Value>::new();
+            let key = generation_result.mandatory_project_key.underlying.clone();
             let (key, value) = self
-                .read_or_fault(&key.underlying, None)
+                .read_or_fault(&key, None)
                 .with_context(|| format!("Reading key {:?} failed", key))?;
+            let key = key.try_into()?;
             let value = value.ok_or_else(|| anyhow::anyhow!("Key {:?} not found", key))?;
-            items.push((key, value));
-        }
+            mandatory_items.insert(key, value);
+            mandatory_items
+        };
+
+        let manifest = ExportManifest {
+            shard_count,
+            mandatory_items,
```

Comment:
Why not have  mandatory_items in export along with the other items? It could save some code duplication